src/System.Net.Security/src/System/Net/SslStreamPal.Unix.cs

```diff
@@ -131,13 +131,17 @@ private static SecurityStatusPal EncryptDecryptHelper(SafeDeleteContext security
             try
             {
                 Interop.Ssl.SslErrorCode errorCode = Interop.Ssl.SslErrorCode.SSL_ERROR_NONE;
-
-
                 SafeSslHandle scHandle = securityContext.SslContext;
 
-                resultSize = encrypt ?
-                    Interop.OpenSsl.Encrypt(scHandle, buffer, offset, size, out errorCode) :
-                    Interop.OpenSsl.Decrypt(scHandle, buffer, size, out errorCode);
+                if (encrypt)
+                {
+                    resultSize = Interop.OpenSsl.Encrypt(scHandle, buffer, offset, size, out errorCode);
+                }
+                else
+                {
+                    Debug.Assert(offset == 0, "Expected offset 0 when decrypting");
```

Comment:
My only question is "is asserting enough?".  If it's really guaranteed, that's fine (like, the calling method always passes 0); otherwise there's "honor it" and "throw".